conans/client/command.py

```diff
@@ -71,6 +72,27 @@ def __call__(self, parser, namespace, values, option_string=None):
         setattr(namespace, self.dest, values)
 
 
+class ConanFilePath(argparse.Action):
```

Comment:
I think that it is the responsibility of the command line to provide the API with a path to a `conanfile.py` (valid inputs to CLI are the path to the conanfile or the path to directory, also valid is a relative path). I think that the `conan_api` should require an absolute path to a `conanfile.py`, there is no need to pass both: the temptative path to the conanfile and also the current working directory.

There is an issue for this: #4157, please, let's move this conversation there and decide where to unify this logic: command vs conan_api. I will mark this PR to depend on that issue. Maybe it should be moved to triaging stage.

Reply:
I think that we have missed the point here. I think there is no bussiness logic here, the `conan_api` expect a path to a `conanfile.py` and the CLI is providing it (handling the possible relative_path and the missing `conanfile.py` ending). I think that this logic does not belong to the conan api, moving it there will offuscate the API with a complex docstring _"target: this argument must be a path to a conanfile.py or to a folder containing one. Also, if given a relative folder, it will be computed from the path where the calling application is running"_ versus a simple _"target: path to a conanfile.py"_. I think it is more aligned with the last comments in #4157, isn't it?

Reply:
Ok, I will move all the data needed inside `conan_api`, but it is against my criteria. Take into account that the help for this argument is _"Path to the package folder in the user workspace"_, if only I had written _"Path to the conanfile.py in the user workspace"_ then it would be ok to keep it in this class, wouldn't it?